pkg/server/commands/reverseexpand/reverse_expand.go

```diff
@@ -171,14 +172,23 @@ func NewResolutionMetadata() *ResolutionMetadata {
 
 // Execute yields all the objects of the provided objectType that the given user has, possibly, a specific relation with
 // and sends those objects to resultChan. It MUST guarantee no duplicate objects sent.
+//
+// If an error is encountered before resolving all objects the provided channel will not be closed and Execute will send
+// the error on the channel promptly and return. Otherwise, Execute will yield all of the objects on the provided channel
+// and then close the channel to signal that it is done.
 func (c *ReverseExpandQuery) Execute(
 	ctx context.Context,
 	req *ReverseExpandRequest,
 	resultChan chan<- *ReverseExpandResult,
 	resolutionMetadata *ResolutionMetadata,
-) error {
-	defer close(resultChan)
-	return c.execute(ctx, req, resultChan, nil, resolutionMetadata)
+) {
+	err := c.execute(ctx, req, resultChan, nil, resolutionMetadata)
+	if err != nil {
+		resultChan <- &ReverseExpandResult{Err: err}
```

Comment:
I ran `BenchmarkOpenFGAServer/BenchmarkPostgresDatastore/BenchmarkListObjects/oneResult` manually with a few print statements added. This line becomes blocked because `resultChan` is only of size 1. If we increase it to size 100, the benchmark runs much faster

Reply:
I checked out `v1.3.4` and tested this and it also occasionally fails, so I believe this is unrelated to this PR directly. We should create an issue and track separately.

Reply:
But then the title of the PR "promptly return if max results are met before deadline" is misleading 🤔

Reply:
Why? In `v1.3.4` we are not promptly returning. We wait all the way up to the deadline. In this PR we do promptly return as soon as we've been cancelled, the deadline has been reached, and/or we found all remaining results.